### PR TITLE
Static build

### PR DESCRIPTION
**Pull request by uooq**
Support for generating a static version of the site

The top level README has a section on how to generate the static pages & how to check them to see if they work.

### CODE REVIEW

**Comment by birkir** [comment]
Cool! I had an idea to use just wget to crawl the entire site, I'm not sure if that will work, but worth the shot.

https://www.explainshell.com/explain?cmd=wget+-P+.%2Fstatic+-mpck+--wait+1+-E+http%3A%2F%2Flocalhost%3A3000

File: server/middleware/errorHandlers.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ const errorHandlersMiddleware = [
    * have this backup for paths not handled by the react middleware. For
    * example you may bind a /api path to express.
    */
-  function notFoundMiddlware(req, res, next) {
+  function notFoundMiddlware(err, req, res, next) {
+    console.log('404 error middleware');
```
Comment (uooq) [inline server/middleware/errorHandlers.js]: oops, this should be removed.

**Comment by uooq** [comment]
@birkir I like your idea but how would we know how far to crawl with respect to dynamic routes? 

I'm thinking that we could instantiate MainApp at build time and inspect the child structure, etc. Might even be able to ascertain the 404 by finding a route with no path attribute, etc. But it's definitely a little down-in-the-weeds & would break if we switch routers, etc. 

I'm open to suggestions... Your approach seems more resilient because it checks from the outside... the only thing I can't figure out is dynamic vs. static route detection...

**Review by finnursig — APPROVED** [review]
:+1:

**Pull request merged**